### PR TITLE
Add listing page and add new option to make collection can more customized

### LINKED ISSUE #144: Make collection random on homepage or recommendation system
Make the homepage to show the random of collection.

### LINKED ISSUE #145: Make view for listing page
Make the view for listing page that show all collection alphabetically.

### LINKED ISSUE #146: Make the mockup for listing page
Make the mockup page for listing page to bind with the view in #145.

### LINKED ISSUE #147: Make the listing page
Make the page for listing all collection that is available in website.

Checklist:
- [x] #145 
- [x] #146

### PR DESCRIPTION
**Pull request by HelloYeew**
Hi! There is three and more things on this PRs.

# Add listing page

Close #145 and Close #146 and Close #147

image

The listing page use the same code as the old homepage but I make the listing page to order alphabetically.

# Revamp homepage

Close #144

image

Now homepage are not show all collection, but it will show the collection from personalization. Currently I add 2 parts to homepage. (Latest created collections and random collection)

# New collection customization

image

## Allow Comments

- Now collection owner can turn off comment section on adding beatmap. As the form said, when the comment is not allowed, the comment part in beatmap card will not show too.
- I set on comment form as not required now, so even collection owner turn on allowed comment user can not add comment too.

image

## Background Alignment

I want to make the beatmap header picture to be more customizable so I add this customization to this. Ths will change the value of `background-align` on header.

# Revamp the redirect link when create the collection

Now when create collection, user will redirect to the collectionp page instead of homepage.

# Fix the bug in osu! ID regressed from #181

image

The osu! username is not go down (it's in the same line as username). Now it's fix.

### CODE REVIEW

**Review by Siraphop4Nene — APPROVED** [review]
Nice work kub.

File: beatmap_collections/views.py [hunk 1]
```diff
@@ -15,12 +15,47 @@ def random_hero_image():
...
+    for i in range(4):
+        # Get latest 4 created collection from database.
+        try:
+            latest_added.append(Collection.objects.order_by('-create_date')[i])
+        except IndexError:
+            # If there is no more collection in database, just break the loop.
+            break
+    collection_count = Collection.objects.count()
+    if collection_count < 4:
+        # To escape the error, if the collection count is less than 4 -> return all collection
```
Comment (pontakornth) [inline beatmap_collections/views.py:29]: You might want to use `Collection.objects.order_by('-create_date')[:4]` instead.  There is no problem even there are less than 4 collections.

File: beatmap_collections/views.py [hunk 2]
```diff
@@ -15,12 +15,47 @@ def random_hero_image():
...
+        random_collection = Collection.objects.all()
+    else:
+        random_collection = []
+        random_count = 0
+        while random_count < 4:
+            random_collection_object = Collection.objects.all()[random.randint(0, collection_count - 1)]
```
Comment (pontakornth) [inline beatmap_collections/views.py:35]: There is a `sample` method in random package. You might want to use that instead.

**Review by pontakornth — CHANGES_REQUESTED** [review]
Also, please add tests in `listing` view. You can create multiple collections and see if it is sorted by name or not.

**Review by SainTurDaY27 — APPROVED** [review]
Good!

**Review by some-random-idiot — APPROVED** [review]

**Review by pontakornth — APPROVED** [review]
We discussed these changes. I approve.

**Pull request merged**